oci/pull.bzl

```diff
@@ -103,7 +103,21 @@ _PLATFORM_TO_BAZEL_CPU = {
     "linux/mips64le": "@platforms//cpu:mips64",
 }
 
-def oci_pull(name, image = None, repository = None, registry = None, platforms = None, digest = None, tag = None, reproducible = True, is_bzlmod = False, config = None, config_path = None):
+def _get_default_toolchain(toolchain = None):
+    if toolchain:
+        return toolchain
+    for r in native.existing_rules().values():
+        if r["kind"] == "crane_repositories":
+            _toolchain = r["name"][:-len(r["platform"]) - len("_crane_")]
+            if not toolchain:
+                toolchain = _toolchain
+            elif toolchain != _toolchain:
+                fail("Multiple oci toolchains found, please specify the toolchain")
+    if not toolchain:
+        fail("No oci toolchain found, please register first", native.existing_rules().values())
```

Comment:
Now the main contents have been completed. But bzlmod's ci failed.

When this macro was called at module_extension context, native.existing_rules() has nothing. And even special the toolchain name, the repository is inaccessible.

`Unable to load package for @@[unknown repo 'oci_crane_darwin_arm64' requested from @@rules_oci~]//:crane: The repository '@@[unknown repo 'oci_crane_darwin_arm64' requested from @@rules_oci~]' could not be resolved: No repository visible as '@oci_crane_darwin_arm64' from repository '@@rules_oci~'`

I read and studied gazelle `go_repository_tools`, but I still don't understand the principle.

@alexeagle @thesayyn could you help me to fix it? Just the last step